tests/visualization_tests/test_contour.py

```diff
@@ -1,47 +1,188 @@
-import itertools
+from io import BytesIO
+import math
+from typing import Any
+from typing import Callable
+from typing import Dict
 from typing import List
 from typing import Optional
 from typing import Tuple
 from typing import Union
 
-from packaging import version
-import plotly
+import numpy as np
 import pytest
 
+from optuna.distributions import BaseDistribution
 from optuna.distributions import CategoricalDistribution
 from optuna.distributions import FloatDistribution
 from optuna.study import create_study
+from optuna.study import Study
+from optuna.visualization import plot_contour as plotly_plot_contour
+from optuna.visualization._contour import _AxisInfo
+from optuna.visualization._contour import _ContourInfo
+from optuna.visualization._contour import _get_contour_info
+from optuna.visualization._contour import _SubContourInfo
+from optuna.visualization._plotly_imports import _imports as plotly_imports
 from optuna.testing.objectives import fail_objective
 from optuna.testing.visualization import prepare_study_with_trials
 from optuna.trial import create_trial
-from optuna.visualization import plot_contour
-from optuna.visualization._contour import _get_contour_subplot
-from optuna.visualization._contour import _get_contour_subplot_info
-from optuna.visualization._utils import _is_reverse_scale
 from optuna.visualization._utils import COLOR_SCALE
+from optuna.visualization.matplotlib import plot_contour as plt_plot_contour
+from optuna.visualization.matplotlib._matplotlib_imports import _imports as plt_imports
+
+
+if plotly_imports.is_successful():
+    from optuna.visualization._plotly_imports import go
+
+
+if plt_imports.is_successful():
+    from optuna.visualization.matplotlib._matplotlib_imports import Axes
+    from optuna.visualization.matplotlib._matplotlib_imports import plt
 
 
 RANGE_TYPE = Union[Tuple[str, str], Tuple[float, float]]
 
+parametrize_plot_contour = pytest.mark.parametrize(
+    "plot_contour", [plotly_plot_contour, plt_plot_contour]
+)
+
 
-def test_target_is_none_and_study_is_multi_obj() -> None:
+def _create_study_with_error_trials() -> Study:
```

Comment:
[nit] since the study contains only one failed trial, so 

```suggestion
def _create_study_with_failed_trial() -> Study:
```

or 

```suggestion
def _create_study_with_only_failed_trial() -> Study:
```

sounds more accurate.